common/src/main/java/gov/cms/ab2d/common/service/JobServiceImpl.java

```diff
@@ -116,6 +116,13 @@ public Job updateJob(Job job) {
     public Resource getResourceForJob(String jobUuid, String fileName) throws MalformedURLException {
         Job job = getJobByJobUuid(jobUuid);
 
+        User user = userService.getCurrentUser();
+        if (!user.equals(job.getUser())) {
+            log.error("User attempted to download a file where they had a valid UUID, but was not logged in as the " +
+                    "user that created the job");
+            throw new InvalidJobAccessException("You don't have permissions to access the job " + jobUuid);
```

Comment:
The problem with this error message text is that it serves as a confirmation to the unauthorized user that a job with such UUID indeed exists. We don't really need to tell them that. Let's just return `Unauthorized`.